rio_tiler_pds/landsat/aws/landsat_collection2.py

```diff
@@ -100,37 +117,34 @@ class LandsatC2L2Reader(MultiBandReader):
 
     _scheme: str = "s3"
     _hostname: str = "usgs-landsat"
-    _prefix: str = "collection02/level-{_processingLevelNum}/standard/{_sensor}/{acquisitionYear}/{path}/{row}/{scene}/{scene}"
+    _prefix: str = "collection02/level-{_processingLevelNum}/standard/{sensor_name}/{acquisitionYear}/{path}/{row}/{scene}/{scene}"
 
     def __attrs_post_init__(self):
         """Fetch productInfo and get bounds."""
         self.scene_params = sceneid_parser(self.sceneid)
 
         processing_level = self.scene_params["processingCorrectionLevel"]
-        sensor = self.scene_params["_sensor"]
+        sensor_name = self.scene_params["sensor_name"]
 
-        # For collection 2, level 2, ETM and TM have the same bands
         if processing_level == "L2SR":
-            if sensor == "oli-tirs":
+            if sensor_name in ["oli-tirs", "oli"]:
```

Comment:
added `oli`, I think it was missing but I'm maybe wrong

Reply:
Maybe an exception should be raised here for `sensor_name == 'mss'`

Reply:
there is no `mss` for level 2

Reply:
Right, but it doesn't hurt to add an exception in case that comes up, because it might mean a bug somewhere.

Reply:
sure! it might be easier to raise an error in `sceneid_parser` then!